### ISSUE TITLE
auto calculated tolerant ratio may not suitable for big region size

### ISSUE THREAD

**Issue by disksing** [tikv/pd#3880]
Currently the tolerant size ratio for balance region is automatically calculated based on region count. If the split size is set to a big value (for example, 16G) on TiKV side, a cluster may have very few region count even if the dataset is not small.

/cc @busyjay @rleungx @nolouch

**Comment by nolouch** [comment]
Does this appear in the big region POC version of TiKV?

**Comment by disksing** [comment]
Yes. It appears when we set the region split size to 16G. (actual region size may range from about 2GB~16GB )

**Comment by rleungx** [comment]
I think we need to reconsider the auto adjustment for the tolerant size ratio under this situation. Maybe it should be based on size instead of count.
